uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -1641,6 +1643,90 @@ sub computeSNR {
     }
 }
 
+=pod
+
+=head3 computeDeepQC($tarchiveID, $tarchive_srcloc, $profile)
+
+Calls the DeepQC app on T1 MNC files to get the probability of passing QC.
+
+INPUTS: tarchive ID, tarchive source location, configuration file (usually prod)
+
+=cut
+
+################################################################
+#######################  computeDeepQC #############################
+################################################################
+sub computeDeepQC {
+    my $this = shift;
+    my ($row, $filename, $fileID, $base, $fullpath, $message);
+    my ($tarchiveID, $upload_id, $profile)= @_;
+    my $data_dir = NeuroDB::DBI::getConfigSetting(
+        $this->{dbhr},'dataDirBasepath'
+    );
+    my $ua = LWP::UserAgent->new;
+    my $json = 'JSON::PP'->new;
+    # note: url must be changed to production version later
+    # note 2: include DeepQC modality in prod file
+    my $t1_scan_type = NeuroDB::DBI::getConfigSetting(
+        $this->{dbhr},'t1_scan_type'
+    );
+    my $acqID_query = "SELECT ID FROM mri_scan_type WHERE Scan_type=?";
+    my $acqID_result = ${$this->{'dbhr'}}->prepare($acqID_query);
+    $acqID_result->execute($t1_scan_type);
+    $row = $acqID_result->fetchrow_hashref();
+    my $t1_scan_result = $row->{'ID'};
+
+    my $file_query = "SELECT FileID, file, AcquisitionProtocolID from files f WHERE f.TarchiveSource=?";
+    if ($this->{debug}) {
+        print $file_query . "\n";
+    }
+    my $minc_file_arr = ${$this->{'dbhr'}}->prepare($file_query);
+    $minc_file_arr->execute($tarchiveID);
+
+    while ($row = $minc_file_arr->fetchrow_hashref()) {
+        $filename = $row->{'file'};
+        my $fileID = $row->{'FileID'};
+        my $acqID = $row->{'AcquisitionProtocolID'};
+        $base = basename($filename);
+        $fullpath = $data_dir . "/" . $filename;
+        if (-e $fullpath) {
+            if ($acqID == $t1_scan_result) {
+                my $url = 'http://127.0.0.1:5000/deepqc/' . $fileID;
```

Comment:
shouldn't the http address be in the Config? As I understand it, this is the url for IBIS but other projects might have different url for the deep QC algo since the algo would be different for each project (at least that is what I understood from our discussion).

Reply:
@cmadjar - http://127.0.0.1:5000/ would just be the localhost address since the app would be hosted on the same server where all the MNC files are. I can add this more explicitly to the config file though so that it can be customizable.

Reply:
I agree that the URL should be in configuration somewhere